stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/EventsParser.kt

```diff
@@ -11,20 +11,18 @@ import okhttp3.Response
 import okhttp3.WebSocket
 
 internal class EventsParser(
-    private val parser: ChatParser
+    private val parser: ChatParser,
+    private val service: ChatSocketService,
 ) : okhttp3.WebSocketListener() {
 
     private var connectionEventReceived = false
     private val logger = ChatLogger.get("Events")
-    private lateinit var service: ChatSocketService
-
-    fun setSocketService(service: ChatSocketService) {
-        this.service = service
-    }
+    private var closedByClient = true
```

Comment:
Nice a quick solution, but we can iterate over this logic. Personally, I don't like having states using boolean flags, there are already two of such. We can use state patter or state machine inside it later